plugins/by-name/tiny-devicons-auto-colors/default.nix

```diff
@@ -0,0 +1,124 @@
+          If not provided, the plugin will fetch highlights from the current theme to generate a color palette.
+        '';
+
+    factors = {
+      lightness = defaultNullOpts.mkNum 1.75 ''
+        Adjust the lightness factor.
+      '';
+      chroma = defaultNullOpts.mkNum 1 ''
+        Adjust the chroma factor.
+      '';
+      hue = defaultNullOpts.mkNum 1.25 ''
+        Adjust the hue factor.
+      '';
+    };
+
+    cache = {
+      enabled = defaultNullOpts.mkBool true ''
+        Enable or disable caching to improve performance.
+      '';
+      path = defaultNullOpts.mkStr (lib.nixvim.literalLua ''vim.fn.stdpath("cache") .. "/tiny-devicons-auto-colors-cache.json"'') ''
+        Path to the cache file.
+      '';
+    };
+
+    precise_search = {
+      enabled = defaultNullOpts.mkBool true ''
+        Enable or disable precise search for better color matching.
+      '';
+      iteration = defaultNullOpts.mkNum 10 ''
+        Number of iterations for precise search.
+      '';
+      precision = defaultNullOpts.mkNum 20 ''
+        Precision level for the search.
+      '';
+      threshold = defaultNullOpts.mkNum 23 ''
+        Threshold to consider a color as a match.
+      '';
+    };
+
+    ignore = defaultNullOpts.mkListOf lib.types.str [ ] ''
+      A list of icon names to ignore.
+    '';
+
+    autoreload = defaultNullOpts.mkBool false ''
+      Automatically reload colors when the colorscheme changes.
+    '';
+  };
+
+  settingsExample = {
+    colors = {
+      red = "#ff0000";
+      green = "#00ff00";
+    };
+    factors = {
+      lightness = 1.5;
+      chroma = 1.2;
+      hue = 1.1;
+    };
+    cache = {
+      enabled = true;
+      path = "/path/to/cache.json";
+    };
+    precise_search = {
+      enabled = true;
+      iteration = 15;
+      precision = 25;
+      threshold = 20;
+    };
+    ignore = [
+      "lua"
+      "vim"
+    ];
+    autoreload = true;
+  };
+
+  # We want to make sure its after either of the dev icons providers
+  callSetup = false;
+  extraConfig = cfg: {
+    assertions = [
+      {
+        assertion =
+          config.plugins.web-devicons.enable
+          || (
+            config.plugins.mini.enable
+            && config.plugins.mini.modules ? icons
+            && config.plugins.mini.mockDevIcons
+          );
+        message = ''Nixvim: The plugin `web-devicons` must be enabled to use `tiny-devicons-auto-colors`'';
+      }
+    ];
+    plugins =
+      let
+        setupCode = ''
+          require('tiny-devicons-auto-colors').setup(${
+            lib.optionalString (cfg ? settings) (lib.nixvim.toLuaObject cfg.settings)
+          })
+        '';
+      in
+      {
+        mini.luaConfig.post = lib.mkIf (
+          config.plugins.mini.enable
+          && config.plugins.mini.modules ? icons
+          && config.plugins.mini.mockDevIcons
+        ) setupCode;
+        web-devicons.luaConfig.post = lib.mkIf config.plugins.web-devicons.enable setupCode;
```

Comment:
Q: are you happy with this or were you planning to explore a `mkOrder` based solution?

Reply:
I can switch to using that I think, essentially we would set the priority for the icons providers to the same level and set this below that priority. Does that sound correct?

Reply:
Yeah, sounds correct. Obviously, higher number is lower priority.

My only concern it that by setting a priority for these three plugins, they'll _all_ end up in a different position from where they are currently.

Currently they're merged pseudorandomly within the "default" merge order prio.

Reply:
The only thing that gets a little messy, right now, is delegating the mini priority in the current form of the module. The `mini.icons` stuff gets bundled in the same luaConfig as the rest of their modules. Should we isolate that config and move it up or just move all the mini stuff to a higher order?

Reply:
My initial instinct is that it'd be nice to still keep the mini stuff together. I just hope it doesn't cause issues.

This will be no worse than we we migrated everything to `plugins/by-name`, effectively shuffling the merge order.